jcl/src/java.base/share/classes/openj9/internal/tools/attach/target/DiagnosticUtils.java

```diff
@@ -348,6 +532,269 @@ private static DiagnosticProperties doDump(String diagnosticCommand) {
 		return result;
 	}
 
+/*[IF JFR_SUPPORT]*/
+/*[IF JAVA_SPEC_VERSION == 11]*/
+	private static Long convertToBytes(String sizeValue) {
+		long sizeInBytes = 0L;
+
+		/* Extract the numeric part and the size unit */
+		String numericPart = sizeValue.replaceAll("[^0-9]", "");
+
+		/* Convert the unit to lowercase for consistency */
+		String sizeUnit = sizeValue.replaceAll("[0-9]", "").toLowerCase();
+
+		/* Parse the numeric part */
+		long size = Long.parseLong(numericPart);
+
+		/* Convert to bytes based on the unit */
+		switch (sizeUnit) {
+		case "k":
+			sizeInBytes = size * 1_024L;
+			break;
+		case "m":
+			sizeInBytes = size * 1_024L * 1_024L;
+			break;
+		case "g":
+			sizeInBytes = size * 1_024L * 1_024L * 1_024L;
+			break;
+		default:
+			/* No unit or unrecognized unit, assume bytes */
+			sizeInBytes = size;
+			break;
+		}
+		return sizeInBytes;
+	}
+/*[ENDIF] JAVA_SPEC_VERSION == 11 */
+/*[ENDIF] JFR_SUPPORT */
+
+/*[IF JFR_SUPPORT]*/
+/*[IF JAVA_SPEC_VERSION == 11]*/
+	private static Long convertToNanoseconds(String timeValue) {
+		long timeInNano = 0L;
+
+		/* Extract the numeric part and the time unit */
+		String numericPart = timeValue.replaceAll("[^0-9]", "");
+
+		/* Convert the unit to lowercase for consistency */
+		String timeUnit = timeValue.replaceAll("[0-9]", "").toLowerCase();
+
+		/* Parse the numeric part */
+		long time = Long.parseLong(numericPart);
+
+		/* Convert to nanoseconds based on the unit */
+		switch (timeUnit) {
+		case "s":
+			timeInNano = time * 1_000_000_000L;
+			break;
+		case "m":
+			timeInNano = time * 60 * 1_000_000_000L;
+			break;
+		case "h":
+			timeInNano = time * 60 * 60 * 1_000_000_000L;
+			break;
+		case "d":
+			timeInNano = time * 24 * 60 * 60 * 1_000_000_000L;
+			break;
+		default:
+			/* No unit or unrecognized unit, assume nanoseconds */
+			timeInNano = time;
+			break;
+		}
+		return timeInNano;
+	}
+/*[ENDIF] JAVA_SPEC_VERSION == 11 */
+/*[ENDIF] JFR_SUPPORT */
+
+/*[IF JFR_SUPPORT]*/
+/*[IF JAVA_SPEC_VERSION == 11]*/
+	private static Boolean parseBooleanParameter(String paramName, String[] parameters) {
+		for (String param : parameters) {
+			/* Split only on the first "=" */
+			String[] splitParam = param.split("=", 2);
```

Comment:
Rather than `split()`, suggest using `indexOf('=')` and `substring()`.

Reply:
Thanks, Addressed.